beetsplug/gmusic.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2017, Tigran Kostandyan.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Uploads files to Google Play Music"""
+
+from __future__ import print_function
+import os.path
+
+from beets.plugins import BeetsPlugin
+from beets import ui
+from beets import config
+from beets.ui import Subcommand
+from gmusicapi import Musicmanager, Mobileclient
+import gmusicapi.clients
+
+
+# Checks for OAuth2 credentials, if they don't exist - performs authorization
+m = Musicmanager()
+if os.path.isfile(gmusicapi.clients.OAUTH_FILEPATH):
+    m.login()
+else:
+    m.perform_oauth()
+
+
+class Gmusic(BeetsPlugin):
+    def __init__(self):
+        super(Gmusic, self).__init__()
+
+    def commands(self):
+        gupload = Subcommand('gmusic-upload',
+                             help=u'upload your tracks to Google Play Music')
+
+        search = Subcommand('gmusic-songs',
+                            help=u'list of songs in Google Play Music library'
+                            )
+        search.parser.add_option('-t', '--track', dest='track',
+                                 action='store_true',
+                                 help='Search by track name')
+        search.parser.add_option('-a', '--artist', dest='artist',
+                                 action='store_true',
+                                 help='Search by artist')
+        gupload.func = self.upload
+        search.func = self.search
+        return [gupload, search]
+
+    def upload(self, lib, opts, args):
+        items = lib.items(ui.decargs(args))
+        files = [x.path.decode('utf-8') for x in items]
+        ui.print_('Uploading your files...')
+        m.upload(filepaths=files)
+        ui.print_('Your files were successfully added to library')
+
+    def search(self, lib, opts, args):
+        email = config['gmusic']['email'].as_str()
+        password = config['gmusic']['password'].as_str()
```

Comment:
It would be a good idea to mark the password field (and possibly the email field) as redacted. See other plugins that use passwords in the config file for how to do this.